mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/task/SolveMessageDeletedInconsistenciesService.java

```diff
@@ -0,0 +1,120 @@
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.stream.Collectors;
+
+import jakarta.inject.Inject;
+import jakarta.mail.Flags;
+
+import org.apache.james.mailbox.cassandra.ids.CassandraId;
+import org.apache.james.mailbox.cassandra.mail.CassandraDeletedMessageDAO;
+import org.apache.james.mailbox.cassandra.mail.CassandraMailboxDAO;
+import org.apache.james.mailbox.cassandra.mail.CassandraMessageIdDAO;
+import org.apache.james.mailbox.model.Mailbox;
+import org.apache.james.mailbox.model.MessageRange;
+import org.apache.james.task.Task;
+import org.apache.james.util.ReactorUtils;
+import org.apache.james.util.streams.Limit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+
+import reactor.core.publisher.Mono;
+
+public class SolveMessageDeletedInconsistenciesService {
+
+    public record Context(AtomicLong processedMailboxEntries,
+                          ConcurrentLinkedDeque<CassandraId> errors) {
+
+        record Snapshot(long processedMailboxEntries, ImmutableList<CassandraId> errors) {
+        }
+
+        public Context() {
+            this(new AtomicLong(0), new ConcurrentLinkedDeque<>());
+        }
+
+        void incrementProcessedMailboxEntries() {
+            processedMailboxEntries.incrementAndGet();
+        }
+
+        void addError(CassandraId cassandraId) {
+            errors.add(cassandraId);
+        }
+
+        Snapshot snapshot() {
+            return new Snapshot(processedMailboxEntries.get(), ImmutableList.copyOf(errors));
+        }
+    }
+
+    public record RunningOptions(int messagesPerSecond) {
+        public static final RunningOptions DEFAULT = new RunningOptions(100);
+
+        public RunningOptions {
+            Preconditions.checkArgument(messagesPerSecond > 0, "'messagesPerSecond' must be strictly positive");
+        }
+    }
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SolveMessageDeletedInconsistenciesService.class);
+    private static final Duration PERIOD = Duration.ofSeconds(1);
+    private final CassandraMessageIdDAO messageIdDAO;
+    private final CassandraDeletedMessageDAO deletedMessageDAO;
+    private final CassandraMailboxDAO mailboxDAO;
+
+    @Inject
+    public SolveMessageDeletedInconsistenciesService(CassandraMessageIdDAO messageIdDAO,
+                                                     CassandraDeletedMessageDAO deletedMessageDAO,
+                                                     CassandraMailboxDAO mailboxDAO) {
+        this.messageIdDAO = messageIdDAO;
+        this.deletedMessageDAO = deletedMessageDAO;
+        this.mailboxDAO = mailboxDAO;
+    }
+
+    public Mono<Task.Result> fixInconsistencies(Context context, RunningOptions options) {
+        return mailboxDAO.retrieveAllMailboxes()
+            .transform(ReactorUtils.<Mailbox, Task.Result>throttle()
+                .elements(options.messagesPerSecond())
```

Comment:
The code is currently throtling mailboxes which is not what the intent seemed to be (throttling messages)